src/tests/test_motor.cpp

```diff
@@ -0,0 +1,246 @@
+}
+
+BOOST_AUTO_TEST_CASE(test_set_power) {
+    server_to_mock_pipeline([](Motor& client) -> void {
+        client.set_power(1.0);
+        BOOST_CHECK(client.get_power_status().power_pct == 1.0);
+        client.set_power(0.0);
+        BOOST_CHECK(client.get_power_status().power_pct == 0.0);
+        BOOST_CHECK(!client.get_power_status().is_on);
+        client.set_power(-0.5);
+        BOOST_CHECK(client.get_power_status().power_pct == -0.5);
+        BOOST_CHECK(client.get_power_status().is_on);
+    });
+}
+
+BOOST_AUTO_TEST_CASE(test_go_for) {
+    server_to_mock_pipeline([](Motor& client) -> void {
+        client.go_for(1.0, 1.0);
+        BOOST_CHECK(client.get_position() == 1.0);
+        client.go_for(1.0, 1.5);
+        BOOST_CHECK(client.get_position() == 2.5);
+        client.go_for(1.0, -0.5);
+        BOOST_CHECK(client.get_position() == 2.0);
+    });
+}
+
+BOOST_AUTO_TEST_CASE(test_go_to) {
+    server_to_mock_pipeline([](Motor& client) -> void {
+        client.go_to(1.0, 1.0);
+        BOOST_CHECK(client.get_position() == 1.0);
+        client.go_to(0.1, -1.0);
+        BOOST_CHECK(client.get_position() == -1.0);
+    });
+}
+
+BOOST_AUTO_TEST_CASE(test_reset_zero_position) {
+    server_to_mock_pipeline([](Motor& client) -> void {
+        client.go_to(1.0, 1.0);
+        BOOST_CHECK(client.get_position() == 1.0);
+        client.reset_zero_position(1.5);
+        BOOST_CHECK(client.get_position() == -0.5);
+    });
+}
+
+BOOST_AUTO_TEST_CASE(test_get_properties) {
+    server_to_mock_pipeline(
+        [](Motor& client) -> void { BOOST_CHECK(client.get_properties().position_reporting); });
+}
+
+BOOST_AUTO_TEST_CASE(test_stop_motor) {
+    server_to_mock_pipeline([](Motor& client) -> void {
+        client.set_power(1.0);
+        BOOST_CHECK(client.get_power_status().power_pct == 1.0);
+        BOOST_CHECK(client.get_power_status().is_on);
+        client.stop_motor();
+        BOOST_CHECK(client.get_power_status().power_pct == 0.0);
+        BOOST_CHECK(!client.get_power_status().is_on);
+        // This test is a no-op for now because is_moving will always
+        // return false
+        BOOST_CHECK(!client.is_moving());
+    });
+}
+
+BOOST_AUTO_TEST_CASE(test_do_command) {
+    server_to_mock_pipeline([](Motor& client) -> void {
+        AttributeMap expected = fake_map();
+
+        AttributeMap command = fake_map();
+        AttributeMap result_map = client.do_command(command);
+
+        ProtoType expected_pt = *(expected->at(std::string("test")));
+        ProtoType result_pt = *(result_map->at(std::string("test")));
+
+        BOOST_CHECK(result_pt == expected_pt);
+    });
+}
+BOOST_AUTO_TEST_SUITE_END()
+
+BOOST_AUTO_TEST_SUITE(test_motor_service_raw)
+std::shared_ptr<MotorServer> get_motor_server() {
+    auto server = std::make_shared<MotorServer>();
+    server->get_sub_svc()->add(std::string("mock_motor"), MockMotor::get_mock_motor());
+    return server;
+}
+
+BOOST_AUTO_TEST_CASE(test_set_power) {
```

Comment:
@stuqdog We talked about these and I wrote one. However, after creating one test, I am hesitant that this is adding value. I created this test by copy-pasting code from client.cpp (code that is already being run by the tests above). The only difference is that these are able to access grpc::Status which is hidden from the normal invocation.

I'll port over all the other tests if you think it is a good use of time but I am doubtful that it will uncover additional errors or that it will significantly improve dev time if any of this ever did fail.

I think the right think to do is research InProcessChannel to verify it is calling the server correctly and to expose the grpc::Status from these calls so that there is no difference and so that people have an idea if things failed.

Happy to just do it though if you feel strongly on this matter.

Reply:
After reading a bit more about `InProcessChannel` I've come around to your view, I don't think these server tests are doing anything that the preceding tests aren't. Add on the general sadness of the code duplication, and I think these are fine to just get rid of.

Reply:
That is good to hear. Thank you for digging into `InProcessChannel`. We should also test-coverage monitoring to our todo.